mailpoet/assets/js/src/segments/dynamic/dynamic-segments-filters/fields/woocommerce/purchased-with-attribute.tsx

```diff
@@ -90,23 +137,52 @@ export function PurchasedWithAttributeFields({
         dimension="small"
         key="select-segment-product-attribute"
         placeholder={__('Search attributes', 'mailpoet')}
-        options={productAttributesOptions}
-        value={filter((productAttributeOption) => {
-          if (segment.attribute_taxonomy_slug === undefined) {
-            return undefined;
-          }
-          return (
-            segment.attribute_taxonomy_slug === productAttributeOption.value
-          );
-        }, productAttributesOptions)}
+        options={combinedOptions}
+        value={
+          segment.attribute_type === 'local'
+            ? filter((localAttributeOption) => {
+                if (!segment.attribute_local_name) {
+                  return undefined;
+                }
+                return (
+                  `${segment.attribute_local_name}@local` ===
+                  localAttributeOption.value
+                );
+              }, localAttributeOptions)
+            : filter((productAttributeOption) => {
+                if (segment.attribute_taxonomy_slug === undefined) {
+                  return undefined;
+                }
+                return (
+                  segment.attribute_taxonomy_slug ===
+                  productAttributeOption.value
+                );
+              }, productAttributesOptions)
+        }
         onChange={(option: SelectOption): void => {
-          void updateSegmentFilter(
-            {
-              attribute_taxonomy_slug: option.value,
-              attribute_term_ids: [],
-            },
-            filterIndex,
-          );
+          if (localAttributeValues.includes(option.value)) {
+            void updateSegmentFilter(
+              {
+                attribute_type: 'local',
+                attribute_local_name: option.value.replace(/@local$/, ''),
+                attribute_local_values: [],
+                attribute_taxonomy_slug: null,
+                attribute_term_ids: null,
+              },
+              filterIndex,
+            );
+          } else {
+            void updateSegmentFilter(
+              {
+                attribute_type: 'taxonomy',
+                attribute_local_name: null,
+                attribute_local_values: null,
+                attribute_taxonomy_slug: option.value,
+                attribute_term_ids: [],
+              },
+              filterIndex,
+            );
+          }
```

Comment:
Similarly to the previous comment we could move this into a function out of the JSX code.